apps/web/src/components/AssessmentResultsTable/AssessmentResultsTable.tsx

```diff
@@ -0,0 +1,210 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { ColumnDef, createColumnHelper } from "@tanstack/react-table";
+import uniqueId from "lodash/uniqueId";
+
+import { commonMessages, getLocalizedName } from "@gc-digital-talent/i18n";
+import { AssessmentResult } from "@gc-digital-talent/graphql";
+import { notEmpty } from "@gc-digital-talent/helpers";
+import {
+  getAssessmentStepType,
+  getPoolSkillType,
+} from "@gc-digital-talent/i18n/src/messages/localizedConstants";
+
+import Table from "~/components/Table/ResponsiveTable/ResponsiveTable";
+import {
+  AssessmentResultType,
+  AssessmentStep,
+  AssessmentStepType,
+  PoolCandidate,
+  PoolSkill,
+} from "~/api/generated";
+
+import cells from "../Table/cells";
+import { buildColumn, columnStatus } from "./utils";
+import { AssessmentStepResult } from "./types";
+
+const columnHelper = createColumnHelper<AssessmentStepResult>();
+
+interface AssessmentResultsTableProps {
+  poolCandidate: PoolCandidate;
+}
+
+const AssessmentResultsTable = ({
+  poolCandidate,
+}: AssessmentResultsTableProps) => {
+  const intl = useIntl();
+
+  // Get assessment steps from pool
+  const assessmentSteps: Array<AssessmentStep> = React.useMemo(() => {
+    const steps = poolCandidate.pool.assessmentSteps ?? [];
+    return steps.filter(notEmpty);
+  }, [poolCandidate.pool.assessmentSteps]);
+
+  // Get all pool skills from assessment steps and remove duplicates
+  const poolSkills: Array<PoolSkill> = React.useMemo(() => {
+    const ps = poolCandidate.pool.poolSkills ?? [];
+    return ps.filter(notEmpty);
+  }, [poolCandidate.pool.poolSkills]);
+
+  // create assessment step results from pool skills
+  const assessmentResults: Array<AssessmentResult> = React.useMemo(() => {
+    const results = poolCandidate.assessmentResults ?? [];
+    return results.filter(notEmpty);
+  }, [poolCandidate.assessmentResults]);
```

Comment:
Sorry, getting in early to let you know that these could all be simplified using `unpackMaybes`. I.e

```tsx
const assessmentSteps = unpackMaybes(poolCandidate?.pool?.assessmentSteps);
const poolSkills = unpackMaybes(poolCandidate?.pool?.poolSkills);
const assessmentResults= unpackMaybes(poolCandidate?.assessmentResults);
```

Reply:
Thanks Eric! Fixed by [a5d5e7a](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9324/commits/a5d5e7a18e43fb4dd88bb7f2523e99733fb416f3)

Reply:
Oh, i don't think we need to memoization here. The compute time to memoize is probably more than the filtering of `undefined`